### ISSUE TITLE
UNet1DModel Input Tensor shape

### ISSUE THREAD

**Issue by SvenGroen** [huggingface/diffusers#1632]
Hello,

I am trying to use a custom [Unet1DModel](https://github.com/huggingface/diffusers/blob/3faf204c496ffed9ad48862874db7031bbe2674e/src/diffusers/models/unet_1d.py). I got it running with the[ code that is used for testing](https://github.com/huggingface/diffusers/blob/3faf204c496ffed9ad48862874db7031bbe2674e/tests/models/test_models_unet_1d.py):

```
from diffusers.utils import floats_tensor, slow, torch_device
def dummy_input():
    batch_size = 1000
    num_features = 14
    seq_len = 16

    noise = floats_tensor((batch_size, num_features, seq_len)).to(torch_device)
    time_step = torch.tensor([10] * batch_size).to(torch_device)

    return {"sample": noise, "timestep": time_step}
def prepare_init_args_and_inputs_for_common():
        init_dict = {
            "block_out_channels": (32, 64, 128),
            "in_channels": 14,
            "out_channels": 14,
            "time_embedding_type": "positional",
            "use_timestep_embedding": True,
            "flip_sin_to_cos": False,
            "freq_shift": 1.0,
            "out_block_type": "OutConv1DBlock",
            "mid_block_type": "MidResTemporalBlock1D",
            "down_block_types": ( "DownResnetBlock1D", "DownResnetBlock1D", "DownResnetBlock1D"),
            "up_block_types": ( "UpResnetBlock1D", "UpResnetBlock1D"),
            "act_fn": "mish",
        }
        inputs_dict = dummy_input()
        return init_dict, inputs_dict

init_dict, inputs_dict = prepare_init_args_and_inputs_for_common()
model = UNet1DModel(**init_dict)
model.to(torch_device)
# model.eval()

with torch.no_grad():
    output = model(**inputs_dict)

    if isinstance(output, dict):
        output = output.sample


print(output.shape) #torch.Size([1000, 14, 16])
```

I struggle to set `seq_len `correctly without getting any error. It works for multiples of 8 but not for other lengths like e.g. 28.

I am trying to use the Unet1D on a tabular dataset of shape `[bs, 1, numb_columns]` and `numb_columns` is not a multiple of 8.

(inside the [docs ](https://huggingface.co/docs/diffusers/api/models#diffusers.UNet1DModel) num_features is called sample_size, and seq_len is called num_channels)

**Comment by mayank64ce** [comment]
hey. Just wanted to know how did you fix this issue? I am trying with sequences of length 63 but am not able to get it to work.

**Comment by SvenGroen** [comment]
@mayank64ce if I remember correctly, I only got it running by adding zeros to my input so it has a length of something that is a multiple of 8 (so in your case 1 additional value to length 64). So basically adding a dummy value that stays the same and has no influence on the output.

I think, in theory, the network should be able to learn that this value does not carry any meaning and should learn to ignore it.

But I only did a couple of tests with this and did not end up using the model...so make sure to try it out yourself and see if it works.
